src/java.base/share/classes/sun/security/ssl/ServerHello.java

```diff
@@ -565,6 +565,32 @@ public byte[] produce(ConnectionContext context,
                     clientHello);
             shc.serverHelloRandom = shm.serverRandom;
 
+            // For key derivation, we will either use the traditional Key
+            // Agreement (KA) model or the Key Encapsulation Mechanism (KEM)
+            // model, depending on what key exchange group is used.
+            //
+            // In JSSE for KA flow, the server usually generates its key
```

Comment:
Minor nit here.  For KA flows, the server first receives the client's share, THEN generates its key share.   Then we finally come here.   Just reverse the order and you should be good.